tests/Aql/EnumerateCollectionExecutorTest.cpp

```diff
@@ -156,6 +155,313 @@ TEST_F(EnumerateCollectionExecutorTestNoRowsUpstream, the_producer_waits) {
+              projections, coveringIndexAttributePositions, useRawPointers, random) {
+    // fakedQuery = _server.createFakeQuery(true, "return 1");
+    // engine =std::make_unique<ExecutionEngine>(*fakedQuery, SerializationFormat::SHADOWROWS);
+    // fakedQuery->setEngine(engine.release());
+    // vocbase.createCollection(json->slice());
+    // aqlCollection = aqlCollection("UnitTestCollection", &vocbase,
+    // arangodb::AccessMode::Type::READ); fakedQuery->setEngine(engine);
+  }
+
+  auto makeInfos(RegisterId outputRegister = 0, RegisterId nrInputRegister = 1,
+                 RegisterId nrOutputRegister = 1,
+                 std::unordered_set<RegisterId> regToClear = {},
+                 std::unordered_set<RegisterId> regToKeep = {})
+      -> EnumerateCollectionExecutorInfos {
+    EnumerateCollectionExecutorInfos infos{
+        outputRegister, nrInputRegister, nrOutputRegister,
+        regToClear,     regToKeep,       engine,
+        &aqlCollection, &outVariable,    varUsedLater,
+        nullptr,        projections,     coveringIndexAttributePositions,
+        useRawPointers, random};
+    block = SharedAqlItemBlockPtr{new AqlItemBlock(itemBlockManager, 1000, nrOutputRegister)};
+    return infos;
+  }
+
+  // insert amount of documents into the vocbase
+  VPackOptions* insertDocuments(size_t amount, std::vector<std::string>& queryResults) {
+    // TODO: Can be optimized to not use AQL INSERT (trx object directly instead)
+    std::string insertQuery =
+        R"aql(INSERT {_key: "testee1", value: 1, sortValue: 1, nestedObject: {value: 1} } INTO UnitTestCollection RETURN NEW)aql";
+    SCOPED_TRACE(insertQuery);
+    auto queryRes = arangodb::tests::executeQuery(vocbase, insertQuery, {});
+    queryResults.push_back(queryRes.data.get()->slice().at(0).toJson(queryRes.context->getVPackOptions()));
+
+    for (size_t i = 2; i <= amount; i++) {
+      std::string insertQueryPart1 = R"aql(INSERT {_key: "testee)aql";
+      std::string insertQueryPart2 = std::to_string(i);
+      std::string insertQueryPart3 =
+          R"(", value: 1, sortValue: 1, nestedObject: {value: 1} } INTO UnitTestCollection RETURN NEW)";
+      std::string finalQuery = insertQueryPart1 + insertQueryPart2 + insertQueryPart3;
+      SCOPED_TRACE(finalQuery);
+      auto queryResInner = arangodb::tests::executeQuery(vocbase, finalQuery, {});
+      queryResults.push_back(queryResInner.data.get()->slice().at(0).toJson(queryResInner.context->getVPackOptions()));
+    }
+
+    return queryRes.context->getVPackOptions();
+  }
+
+  MatrixBuilder<1> buildQueryDocumentsMatrix(std::vector<std::string>& queryResults) {
+    size_t documentAmount = queryResults.size();
+    MatrixBuilder<1> matrix;
+    matrix.reserve(documentAmount);
+    for (auto const& documentStr : queryResults) {
+      const char* cstr = documentStr.c_str();
+      matrix.emplace_back(RowBuilder<1>{cstr});
+    }
+
+    return matrix;
+  }
+};
+
```

Comment:
Can you please add a comment why the tests are disabled?
And what needs to be done to fix them?

Also please use:
DISABLED_produce_all_documents
instead of comments. to disable the test.